ext/storage_sources/azure_store/azure_connection.cpp

```diff
@@ -31,16 +31,35 @@
 #include <azure/core.hpp>
 #include <azure/storage/blobs.hpp>
 
+#include <stdlib.h>
+#include <iostream>
+#include <vector>
+#include <string>
+
 azure_connection::azure_connection(const std::string &bucket_name, const std::string &obj_prefix)
     : _azure_client(Azure::Storage::Blobs::BlobContainerClient::CreateFromConnectionString(
         std::getenv("AZURE_STORAGE_CONNECTION_STRING"), bucket_name)),
       _bucket_name(bucket_name), _object_prefix(obj_prefix)
 {
 }
 
+/*
+ * Build a list of all of the objects in the bucket
+ */
 int
-azure_connection::list_objects(std::vector<std::string> &objects) const
+azure_connection::list_objects(
+  const std::string &prefix, std::vector<std::string> &objects, bool list_single) const
 {
+    Azure::Storage::Blobs::ListBlobsOptions blob_parameters;
+    blob_parameters.Prefix = prefix;
+    if (list_single)
+        blob_parameters.PageSizeHint = 1;
```

Comment:
Have we tested this?

Reply:
yes :) passed unit tests we ran on lachlans machine